upup/pkg/fi/utils/hash.go

```diff
@@ -32,3 +34,31 @@ func HashString(s string) (string, error) {
 	shaStr := hex.EncodeToString(sha) // String representation
 	return shaStr, nil
 }
+
+const (
+	letterBytes   = "abcdefghijklmnopqrstuvwxyz0123456789"
+	letterIdxBits = 6                    // 6 bits to represent a letter index
+	letterIdxMask = 1<<letterIdxBits - 1 // All 1-bits, as many as letterIdxBits
+	letterIdxMax  = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
+)
+
+// RandomString generates a random string of n length
+// source https://stackoverflow.com/questions/22892120/how-to-generate-a-random-string-of-a-fixed-length-in-go
+func RandomString(n int) string {
```

Comment:
upup/pkg/fi/secrets.go has:

```
func CreateSecret() (*Secret, error) {
	data := make([]byte, 128)
	_, err := crypto_rand.Read(data)
	if err != nil {
		return nil, fmt.Errorf("error reading crypto_rand: %v", err)
	}

	s := base64.StdEncoding.EncodeToString(data)
	r := strings.NewReplacer("+", "", "=", "", "/", "")
	s = r.Replace(s)
	s = s[:32]

	return &Secret{
		Data: []byte(s),
	}, nil
}
```

I know crypto_rand is overkill, but it's probably an easier API for us to use.

Also... apparently I used strings.Replacer in the past!